src/BrickbattleWeapons/MainModule/ServerModules/slices/BrickbattleWeaponSlice.lua

```diff
@@ -0,0 +1,34 @@
+--!strict
+local Reflex = require("Packages/reflex")
+local Immut = require("Packages/immut")
+local Table = require("src/BrickbattleWeapons/MainModule/ServerModules/slices/BrickbattleWeaponTables")
+
+export type BrickbattleWeaponStateActions = {
+    setStatState: (playerName: string, numb: number, stat: string) -> (),
+    removeStatState: (playerName: string, stat: string) -> (),
+}
+
+type Data = {
+    [string]: Table.BrickbattleWeaponTable
+}
+local initialState: Data = {}
+local BrickbattleWeaponsStateSlice = Reflex.createProducer(initialState, {
+    setStatState = function(state, playerName: string, numb: number, stat: string)
```

Comment:
this function is used to create a copy of that projectile data table for a new player? have you considered how other developers will add their own copies of projectile data, read from it, etc.?

Reply:
moreover: let's be sure we add a layer of abstraction (our own API) that hides the implementation details from 
other developers. i.e., they shouldn't need to know we use Reflex or that it exists